### PR TITLE
SingleparticlePR

### PR DESCRIPTION
**Pull request by MiryamMV**
Continuation of the SingleParticleGenerator PR

### CODE REVIEW

File: source/tests/utils/RandomUtilsTest.cc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    if (phi_max < phi_min) {
+      std::swap(phi_max,phi_min);
+    }
+
+    //auto direction = nexus::Direction(costheta_min,costheta_max,phi_min,phi_max);
+    G4ThreeVector direction = nexus::Direction(costheta_min,costheta_max,phi_min,phi_max);
+    G4double costheta_test = direction.z();
+    G4double phi_test;
+
```
Comment (paolafer) [inline source/tests/utils/RandomUtilsTest.cc:32]: Why don't you use directly `std::atan2()`? Then `phi_test` is simply `std::atan2(direction.y(), direction.x())` and you don't need the `if/else` statement.

**Comment by jacg** [comment]
@MiryamMV Are you getting automatic emails about the failure of the GitHub Actions associated with this PR?

**Comment by MiryamMV** [comment]
> @MiryamMV Are you getting automatic emails about the failure of the GitHub Actions associated with this PR?

Yes

**Review by paolafer — APPROVED** [review]
This PR adds the possibility of generating a random direction limited to a solid angle. A dedicated function has been created and its functionality tested.
A function to generate a random value of energy in a range has also been extracted to be made of general use. I approve this PR.

**Pull request merged**